lib/Ohmbrewer_Relay.cpp

```diff
@@ -110,6 +78,38 @@ Ohmbrewer::Relay::~Relay() {
 }
 
 /**
+ * @param relayPins - controlPin always first in <list>
+ *  controlPin - The Control pin - Data/speed/power level Digital pin number X.
+ *  powerPin - The power pin - on/off line. Digital pin number X.
+ */
+void Ohmbrewer::Relay::initRelay(std::list<int>* relayPins) {
+    int size = relayPins->size();
+    int powerPin = -1;
+    int controlPin = -1;
+    if  (size == 2){ // switch and relay
+        controlPin = relayPins->front();
+        powerPin = relayPins->back();
+    }else if (size == 1){ //relay only
+        controlPin = relayPins->front();
+    }else {//publish error
+        Ohmbrewer::Publisher::publish_map_t pMap;
+        Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
+        pMap[String("list_check_relay")] = String("improperly formed input - Relay(int, int<list>)");
```

Comment:
Since you're moving all this stuff into a cleaner function anyway, you should probably use the simpler Publisher constructor so you don't have to have so many lines.
You also shouldn't have to create it as a pointer (or, subsequently, mess with delete).